azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/core/AbstractCredentialRetriever.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.tools.auth.core;
+
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+import com.azure.identity.CredentialUnavailableException;
+import com.microsoft.azure.AzureEnvironment;
+import com.microsoft.azure.tools.auth.exception.LoginFailureException;
+import com.microsoft.azure.tools.auth.model.AzureCredentialWrapper;
+import lombok.Setter;
+import rx.Single;
+
+public abstract class AbstractCredentialRetriever implements ICredentialRetriever {
+
+    @Setter
+    protected AzureEnvironment env;
```

Comment:
Can we remove @Setter ？

protected --> I think it only used in its sub classes. so I guess maybe we can remove a public @Setter.

1.